libazureinit/src/provision/user.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::process::Command;
+
+use tracing::instrument;
+
+use crate::error::Error;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Provisioner {
+    Useradd,
+    #[cfg(test)]
+    FakeUseradd,
+}
+
+impl Provisioner {
+    pub(crate) fn create(self, name: impl AsRef<str>) -> Result<(), Error> {
+        match self {
+            Self::Useradd => useradd(name.as_ref()),
+            #[cfg(test)]
+            Self::FakeUseradd => Ok(()),
+        }
+    }
+}
+
+#[instrument(skip_all)]
+fn useradd(name: &str) -> Result<(), Error> {
+    let path_useradd = env!("PATH_USERADD");
+    let home_path = format!("/home/{name}");
+
+    let status = Command::new(path_useradd)
+                    .arg(name)
+                    .arg("--comment")
+                    .arg(
+                      "Provisioning agent created this user based on username provided in IMDS",
+                    )
+                    .arg("--groups")
+                    .arg("adm,audio,cdrom,dialout,dip,floppy,lxd,netdev,plugdev,sudo,video")
+                    .arg("-d")
+                    .arg(home_path.clone())
```

Comment:
In case of building with Rust 1.70.0, stable 12 months ago, which will be introduced in https://github.com/Azure/azure-init/pull/94, clippy prints out the warning:

```
warning: redundant clone
  --> libazureinit/src/provision/user.rs:41:35
   |
41 |                     .arg(home_path.clone())
   |                                   ^^^^^^^^ help: remove this
   |
note: this value is dropped without further use
  --> libazureinit/src/provision/user.rs:41:26
   |
41 |                     .arg(home_path.clone())
   |                          ^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
   = note: `#[warn(clippy::redundant_clone)]` on by default
```

Reply:
I'm really surprised that clippy _isn't_ warning about this on newer versions...

Reply:
> I'm really surprised that clippy _isn't_ warning about this on newer versions...

Starting from Rust clippy [1.71](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-171), `redundant clone` has been [moved](https://github.com/rust-lang/rust-clippy/pull/10873) to `nursery`(a.k.a. under development), because of a bunch of false positives in 1.70. That's the reason.